spec/requests/users/companies_spec.rb

```diff
@@ -0,0 +1,70 @@
+require "rails_helper"
+
+RSpec.describe "Companies", type: :request do
+  include_context "when user signed in (controllers)"
+
+  let(:valid_attributes) { { official_name: "Khusaenov Industries" } }
+
+  describe "GET /new" do
+    context "when user doesn't have company" do
+      it "renders a successful response" do
+        get new_users_company_path
+        expect(response).to be_successful
+      end
+    end
+
+    context "when user writer in some company" do
+      before do
+        create :company_member, user: user, role: :writer
+        get new_users_company_path
+      end
+
+      it "renders a successful response" do
+        expect(response).to be_successful
+      end
+    end
+
+    context "when user have his own company" do
+      before do
+        create :company_member, user: user, role: :owner
+        get new_users_company_path
+      end
+
+      it "redirects to the blog page" do
+        expect(response).to redirect_to(root_path)
+      end
+
+      it "shows reason" do
+        follow_redirect!
+        expect(response.body).to include("Not Authorized")
```

Comment:
Should we deduce a more understandable error? For example, "you cannot create a second company".

Reply:
Yeah, I want it too. Only I haven't figured out how to modify this error message. 😢

Reply:
I did it using this article https://actionpolicy.evilmartians.io/#/i18n. I can help you if you want. We can do it in a separate PR.

Reply:
I add issue for it: #56 :raised_hands: